vars/checkDocumentationPullRequests.groovy

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+/**
+ * Library to check documentation pull requests per release.
+ * @param Map args = [:] args A map of the following parameters
+ * @param args.version <required> - Release version to track the documentation PRs for.
+ */
+void call(Map args = [:]) {
+    // Validate Parameters
+    validateParameters(args)
+    def versionTokenize = args.version.tokenize('-')
+    // Qualifiers are not a part of the labels in GitHub. Ignoring it.
+    def version = versionTokenize[0]
+
+    withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+        def openPRs = sh(
+                script: "gh pr list --repo opensearch-project/documentation-website --state open --label v${version} -S \"-label:\\\"6 - Done but waiting to merge\\\"\" --json url --jq '.[].url'",
+                returnStdout: true
+        )
+        if (openPRs) {
+            echo("Documentation pull requests pending to be merged: \n${openPRs}")
+        } else {
+            echo("No open pull requests found!")
+        }
+    }
+}
```

Comment:
What is the case if someone didnt even open a PR or issue?
Is this lib purely check any existing PRs not merged?

Thanks.

Reply:
Yes! Exit criteria only needs check that. Entrance criteria checks issues and linked PRs. We need to meet entrance criteria before exit is checked by design

Reply:
good point, can you check with documentation team, if they have any mechanism to track PRs that are supposed to have an open doc PR but have not done so?

Reply:
Until all PRs are up, the entrance criteria are not met. The exit criteria assume that the entrance criteria have been met, but to make sure, we can definitely add logic to run both queries (entrance and exit) on exit.

Reply:
I believe what @rishabh6788 wants to know is if the developers completely forget to raise even an issue in doc repo to add documentation, is there a mechanism to track that?

Reply:
Hi @rishabh6788 
Can we proceed with this? Will follow up with documentation team but even if there is a way to track those issues I believe it should be an entrance criteria and not exit criteria.

Reply:
Confirmed with @kolchfa-aws that we do not have a mechanism to track if the feature/bug should have a documentation issue. I see https://github.com/opensearch-project/k-NN/blob/main/.github/workflows/create-documentation-issue.yml but looks like it was never run.